src/storage/segment.cc

```diff
@@ -401,6 +469,39 @@ void Segment::GcEntryFreeList(uint64_t version, uint64_t& gc_idx_cnt, uint64_t&
     }
 }
 
+void Segment::FreeTsEntry(::openmldb::base::Node<uint64_t, DataBlock*>* ts_entry_node,
+                          uint64_t& gc_idx_cnt, uint64_t& gc_record_cnt,
+                          uint64_t& gc_record_byte_size) {
+    if (ts_entry_node == NULL) {
+        return;
+    }
+    if (ts_entry_node->GetValue()->dim_cnt_down <= 1) {
+        delete ts_entry_node->GetValue();
```

Comment:
update `gc_record_byte_size` and `gc_record_cnt`. refer https://github.com/4paradigm/OpenMLDB/blob/main/src/storage/segment.cc#L301